nncf/torch/graph/patterns.py

```diff
@@ -11,27 +11,131 @@
+    sigmoid_node = pattern.add_node(['sigmoid'])
+    mul_node = pattern.add_node(['__mul__'])
 
-ACTIVATIONS = RELU | NON_RELU_ACTIVATIONS
+    pattern.add_edge(linear_ops_node, bn_node)
+    pattern.add_edge(bn_node, sigmoid_node)
+    pattern.add_edge(linear_ops_node, mul_node)
 
-ANY_BN_ACT_COMBO = BN + ACTIVATIONS | ACTIVATIONS + BN | BN | ACTIVATIONS
+    main_pattern.add_pattern_alternative(pattern)
 
-SINGLE_OPS = ACTIVATIONS | POOLING | N('mean') | N('layer_norm')
+    pattern = GraphPattern()
+    linear_ops_node = pattern.add_node(LINEAR_OPERATIONS)
+    sigmoid_node = pattern.add_node(['sigmoid'])
+    mul_node = pattern.add_node(['__mul__'])
 
-ARITHMETIC = N('__iadd__') | N('__add__') | N('__mul__') | N('__rmul__')
+    pattern.add_edge(linear_ops_node, sigmoid_node)
+    pattern.add_edge(linear_ops_node, mul_node)
 
-ELTWISE_UNIFORM_OPS = BN | RELU | ACTIVATIONS
+    main_pattern.add_pattern_alternative(pattern)
 
-MATMUL = N('matmul') | N('bmm')
+    return main_pattern
+
+
+def create_linear_h_swish_act() -> GraphPattern:
+    main_pattern = GraphPattern()
+
+    pattern = GraphPattern()
+    linear_ops_node = pattern.add_node(LINEAR_OPERATIONS)
+    bn_node = pattern.add_node(BATCH_NORMALIZATION_OPERATIONS)
+    __add__node = pattern.add_node(['__add__'])
+    hardtanh_node = pattern.add_node(['hardtanh'])
+    __truediv__node = pattern.add_node(['__truediv__'])
+    mul_node = pattern.add_node(['__mul__'])
+
+    pattern.add_edge(linear_ops_node, bn_node)
+    pattern.add_edge(bn_node, __add__node)
+    pattern.add_edge(bn_node, mul_node)
+    pattern.add_edge(__add__node, hardtanh_node)
+    pattern.add_edge(hardtanh_node, __truediv__node)
+    pattern.add_edge(__truediv__node, mul_node)
+
+    main_pattern.add_pattern_alternative(pattern)
+
+    pattern = GraphPattern()
+    linear_ops_node = pattern.add_node(LINEAR_OPERATIONS)
+    __add__node = pattern.add_node(['__add__'])
+    hardtanh_node = pattern.add_node(['hardtanh'])
+    __truediv__node = pattern.add_node(['__truediv__'])
+    mul_node = pattern.add_node(['__mul__'])
+
+    pattern.add_edge(linear_ops_node, __add__node)
+    pattern.add_edge(linear_ops_node, mul_node)
+    pattern.add_edge(__add__node, hardtanh_node)
+    pattern.add_edge(hardtanh_node, __truediv__node)
+    pattern.add_edge(__truediv__node, mul_node)
+
+    main_pattern.add_pattern_alternative(pattern)
+
+    return main_pattern
+
+
+def create_linear_h_sigmoid_act() -> GraphPattern:
+    pattern = GraphPattern()
+    linear_ops_node = pattern.add_node(LINEAR_OPERATIONS)
+    __add__node = pattern.add_node(['__add__'])
+    hardtanh_node = pattern.add_node(['hardtanh'])
+    __truediv__node = pattern.add_node(['__truediv__'])
+
+    pattern.add_edge(linear_ops_node, __add__node)
+    pattern.add_edge(__add__node, hardtanh_node)
+    pattern.add_edge(hardtanh_node, __truediv__node)
+
+    return pattern
+
+
+@PATTERN_GRAPH.register('graph_pattern_factory')
```

Comment:
What is the purpose of the `PATTERN_GRAPH` registry? Does it store graph patterns? If so, why do you register a factory there?

Reply:
Support. No reason to have a registry for one entity. Can we split this into multiple graphs-patterns and iterate over them when we do the matching?

Reply:
Done; Implement QuantizationIgnorePatterns that stores all registered patterns